pkg/flow/events.go

```diff
@@ -0,0 +1,87 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package flow
+
+import (
+	"strings"
+
+	"github.com/onflow/flow-go-sdk"
+)
+
+const addressLength = 16
+
+type Event struct {
+	Type   string
+	Values map[string]string
+}
+
+type Events []Event
+
+func EventsFromTransaction(tx *flow.TransactionResult) Events {
+	var events Events
+	for _, event := range tx.Events {
+		events = append(events, newEvent(event))
+	}
+
+	return events
+}
+
+func newEvent(event flow.Event) Event {
+	var names []string
+
+	for _, eventType := range event.Value.EventType.Fields {
+		names = append(names, eventType.Identifier)
+	}
+	values := map[string]string{}
+	for id, field := range event.Value.Fields {
+		name := names[id]
+		values[name] = field.String()
+	}
+
+	return Event{
+		Type:   event.Type,
+		Values: values,
+	}
+}
+
+func (e *Events) GetAddress() *flow.Address {
```

Comment:
Not a blocker: I think there's probably best to not include logic specific to one event inside the `Events` type

Reply:
What if there will be others in future? I think there might, but if not where should I put this?